### ISSUE TITLE
ol.geom.Circle is not considered in ol.interaction.Snap, exception is thrown

### ISSUE THREAD

**Issue by hajjimurad** [openlayers/openlayers#6748]
I have a layer based on ol.source.Vector with collection of the following features:

- Point
- LineString
- Polygon
- GeometryCollection (**Circle** + Point)

... and want to add snap interaction while drawing over this layer. But I'm getting the following error

```
VM52171:240 Uncaught TypeError: Cannot read property 'call' of undefined
    at ol.interaction.Snap.writeGeometryCollectionGeometry_ (eval at 65 (http://localhost:8080/app.js:1122:1), <anonymous>:1508:404)
    at ol.interaction.Snap.addFeature (eval at 65 (http://localhost:8080/app.js:1122:1), <anonymous>:1500:289)
    at ol.interaction.Snap.forEachFeatureAdd_ (eval at 65 (http://localhost:8080/app.js:1122:1), <anonymous>:1500:490)
    at Array.forEach (native)
    at ol.interaction.Snap.setMap (eval at 65 (http://localhost:8080/app.js:1122:1), <anonymous>:1505:172)
    at ol.Map.eval (eval at 65 (http://localhost:8080/app.js:1122:1), <anonymous>:598:409)
    at ol.Collection.b (eval at 65 (http://localhost:8080/app.js:1122:1), <anonymous>:128:506)
    at ol.Collection.ol.events.EventTarget.dispatchEvent (eval at 65 (http://localhost:8080/app.js:1122:1), 
```

After investigation of **snap.js** I found that there are SEGMENT_WRITERS_ which are not defined for 'Circle'. And therefore the exception is thrown. 

From my understanding **Circle** is ignored when features are being added to Snap-interaction, but it's not ignored in **writeGeometryCollectionGeometry_** function in snap.js:

```
ol.interaction.Snap.prototype.writeGeometryCollectionGeometry_ = function(feature, geometry) {
  var i, geometries = geometry.getGeometriesArray();
  for (i = 0; i < geometries.length; ++i) {
    this.SEGMENT_WRITERS_[geometries[i].getType()].call(
        this, feature, geometries[i]);
  }
};
```

I guess it has sense to proceed with one of the following approaches: 
- implement SEGMENT_WRITER for **Circle** 
- ignore **Circle** in writeGeometryCollectionGeometry_

**Comment by ahocevar** [comment]
See #4304.

**Comment by hajjimurad** [comment]
Thanks for the quick reply!

I actually don't want to snap to Circles and current behavior is suitable for me. I'm talking only about thrown exception.

Would you accept a pull request, if I make Cirlces ignored in writeGeometryCollectionGeometry_ the same way as it's done in **addFeature** function (snap.js)?

**Comment by ahocevar** [comment]
Yes, please submit a pull request.

**Comment by hajjimurad** [comment]
It's submitted and waiting for approval

**Comment by fredj** [comment]
fixed with #6757
